contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -2988,83 +3061,143 @@ objectproperty_internal(PG_FUNCTION_ARGS)
 			pg_strcasecmp(property, "isschemabound") == 0 ||
 			pg_strcasecmp(property, "execisansinullson") == 0)
 	{
-		if (pg_strcasecmp(type, "P") != 0 && pg_strcasecmp(type, "RF") != 0 &&
-			pg_strcasecmp(type, "V") != 0 && pg_strcasecmp(type, "TR") != 0 &&
-			pg_strcasecmp(type, "FN") != 0 && pg_strcasecmp(type, "IF") != 0 &&
-			pg_strcasecmp(type, "TF") != 0 && pg_strcasecmp(type, "R") != 0)
+		/*
+		 * These properties are only applicable to TSQL_STORED_PROCEDURE, REPLICATION_FILTER_PROCEDURE,
+		 * VIEW, TSQL_DML_TRIGGER, TSQL_SCALAR_FUNCTION, TSQL_INLINE_TABLE_VALUED_FUNCTION, 
+		 * TSQL_TABLE_VALUED_FUNCTION and RULE.
+		 * Hence, return NULL if the object is not from the above types.
+		 */
+		if (type != TSQL_STORED_PROCEDURE && type != REPLICATION_FILTER_PROCEDURE &&
+			type != VIEW && type != TSQL_DML_TRIGGER && type != TSQL_SCALAR_FUNCTION &&
+			type != TSQL_INLINE_TABLE_VALUED_FUNCTION && type != TSQL_TABLE_VALUED_FUNCTION &&
+			type != RULE)
```

Comment:
This could be written more cleanly like
if (!( type == TSQL_STORED_PROCEDURE || type == REPLICATION_FILTER_PROCEDURE ))